### PR TITLE
Fix file time handling on Windows

### PR DESCRIPTION
**Pull request by NicknineTheEagle**
# Description

This fixes getting/setting file modify time on Windows. GetFileTime and SetFileTime operate with UTC timestamps whereas DOS stores local time - and since FileTimeToDosDateTime and DosDateTimeToFileTime do not do any time zone conversion, the current implementations of get_dos_file_time of set_dos_file_time output incorrect time.

I've added conversion between UTC time and local time to fix this.

Note that I did not check POSIX implementation since I don't have the necessary environment. Someone will need to test it to make sure it doesn't have the same problem.

# Release notes

None since this is a change to in-dev code.

# Manual testing

Tested installing Prince of Persia from floppy, timestamps are correctly copied over.

Tested saving in Doom, timestamp is correctly set to current computer time.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] provided the release notes draft (for significant user-facing changes).
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by FeralChild64** [comment]
> I did not check POSIX implementation since I don't have the necessary environment. Someone will need to test it to make sure it doesn't have the same problem

Just checked, looks like the timestamps on Linux are correct (I'm currently using GMT+2 time zone).

File: vs/dosbox.vcxproj [hunk 1]
```diff
@@ -276,7 +276,13 @@ IF %ERRORLEVEL% LSS 8 SET ERRORLEVEL = 0</Command>
...
     <PostBuildEvent>
       <Command>%systemroot%\System32\robocopy ..\contrib\resources "$(OutDir)resources" /mir /xf "meson.build" "*.sh"
-IF %ERRORLEVEL% LSS 8 SET ERRORLEVEL = 0</Command>
+IF %ERRORLEVEL% LSS 8 SET ERRORLEVEL = 0
+rd /s /q D:\Emulators\Dosbox\resources
+del D:\Emulators\Dosbox\*.exe
+del D:\Emulators\Dosbox\*.dll
+copy $(TargetDir)$(TargetName)$(TargetExt) D:\Emulators\Dosbox
+copy $(TargetDir)*.dll D:\Emulators\Dosbox
+xcopy $(TargetDir)resources D:\Emulators\Dosbox\resources /E /I</Command>
```
Comment (weirddan455) [inline vs/dosbox.vcxproj:285]: What is all this?  I assume you didn't mean to commit this so please remove it.  If it's intentional, put this in a separate commit and explain the purpose.

**Review by weirddan455 — COMMENTED** [review]
Good catch.  I tested in Windows to confirm the bug and your fix.  I also tested in Linux and it's already doing the conversions on both read and write so no changes needed to the POSIX layer.

File: vs/dosbox.vcxproj (hunk 1 above)
Comment (NicknineTheEagle) [reply vs/dosbox.vcxproj:285]: Oh, FFS, not again.

**Review by weirddan455 — APPROVED** [review]
Looks good.  Thanks for the fix.

**Comment by weirddan455** [comment]
I just want to ping @johnnovak before merge since we're getting close to release.  IMO this should go in 0.82.0.  This was a regression from 0.81.2.

**Comment by johnnovak** [comment]
> I just want to ping @johnnovak before merge since we're getting close to release. IMO this should go in 0.82.0. This was a regression from 0.81.2.

Yes, we'll include this in 0.820. Nice catch @NicknineTheEagle and thanks for the fix 🎉 Can you please fill out the release notes section then?

**Comment by NicknineTheEagle** [comment]
> > I just want to ping @johnnovak before merge since we're getting close to release. IMO this should go in 0.82.0. This was a regression from 0.81.2.
> 
> Yes, we'll include this in 0.820. Nice catch @NicknineTheEagle and thanks for the fix 🎉 Can you please fill out the release notes section then?

As I already stated, there are no release notes to make since this fixes a regression from 0.81.2.

**Comment by johnnovak** [comment]
> > > I just want to ping @johnnovak before merge since we're getting close to release. IMO this should go in 0.82.0. This was a regression from 0.81.2.
> > 
> > 
> > Yes, we'll include this in 0.820. Nice catch @NicknineTheEagle and thanks for the fix 🎉 Can you please fill out the release notes section then?
> 
> As I already stated, there are no release notes to make since this fixes a regression from 0.81.2 where file timestamps worked correctly.

Yeah, okay, now I got it. Just a tip, maybe be a bit more clear next time, so this would be better:

> None, regressions since 0.81.2

instead of 

> None since this is a change to in-dev code.
